src/Bicep.Cli/Commands/BuildParamsCommand.cs

```diff
@@ -58,49 +58,45 @@ public async Task<int> RunAsync(BuildParamsArguments args)
                 logger.LogWarning(CliResources.ResourceTypesDisclaimerMessage);
             }
 
-            if(bicepInputPath != null && !IsBicepFile(bicepInputPath))
+            if(bicepFileArgPath != null && !IsBicepFile(bicepFileArgPath))
             {
-                throw new CommandLineException($"{bicepInputPath} is not a bicep file");
+                throw new CommandLineException($"{bicepFileArgPath} is not a bicep file");
             }
 
             if(features.ParamsFilesEnabled && IsBicepparamsFile(paramsInputPath)) 
             {
                 var paramsCompilation = await compilationService.CompileAsync(paramsInputPath, args.NoRestore);
 
-                var bicepSemanticModel = GetBicepSemanticModelFromParamsCompilation(paramsCompilation);
+                var paramsSemanticModel = paramsCompilation.GetEntrypointSemanticModel();
 
-                if(bicepSemanticModel != null && bicepInputPath != null) 
-                {   
-                    var bicepFilePath = bicepSemanticModel.Root.FileUri;
-                    
-                    if(!bicepFilePath.Equals(PathHelper.FilePathToFileUrl(bicepInputPath)))
-                    {
-                        throw new CommandLineException($"Bicep file {bicepInputPath} provided with --bicep-file option doesn't match the Bicep file {bicepSemanticModel.Root.Name} referenced by the using declaration in the parameters file");
-                    }
-                }
-
-                if (diagnosticLogger.ErrorCount < 1)
+                if(paramsSemanticModel.Root.TryGetBicepFileSemanticModelViaUsing(out var bicepSemanticModel, out _))
```

Comment:
It would be worth adding a comment here to explain why we don't need to worry about handling a failure (because the semantic analysis done later on will catch this).